#!/usr/bin/env python
PACKAGE = "orbus_interface"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Encoder
gen.add("CPR",       int_t, 0, "[#] An Integer parameter",        1,  0, 65535)

enc_pos = gen.enum([ gen.const("After",  int_t, 0, "After gears"),
                    gen.const("Before", int_t, 1, "Before gears")],
                    "Position")
gen.add("position", int_t, 0, "Position encoder respect to gears", 1, 0, 1, edit_method=enc_pos)

enc_z_index = gen.enum([ gen.const("NO",  int_t, 0, "Without Z index"),
                    gen.const("YES", int_t, 1, "Z index")],
                    "Z-Index")
gen.add("z_index", int_t, 0, "This encoder use the Z index to find 0 position", 0, 0, 1, edit_method=enc_z_index)

enc_channel = gen.enum([ gen.const("ONE",  int_t, 1, "Single channel"),
                    gen.const("TWO", int_t, 2, "Two channels")],
                    "Channels")
gen.add("channels", int_t, 0, "How many channel have your encoder", 2, 1, 2, edit_method=enc_channel)

#Default command
gen.add("Store_in_EEPROM", bool_t  , 0, "Store this data in uNav EEPROM", False)
gen.add("restore_defaults", bool_t  , 0, "Restore to the original configuration", False)

exit(gen.generate(PACKAGE, "unav_node", "UnavEncoder"))
